logdecorator/decorator.py

```diff
@@ -145,7 +145,7 @@ def _do_logging(self, fn: FunctionType, exception: Exception, *args: Any, **kwar
     def execute(self, fn: FunctionType, *args: Any, **kwargs: Any) -> Any:
         try:
             return super().execute(fn, *args, **kwargs)
-        except Exception as e:
+        except BaseException as e:
```

Comment:
Nice catch! Could you also change the type annotations of `on_error` and `_do_logging`, so they require `BaseException` instead of `Exception`?

You can check if everything is alright with `mypy -p logdecorator`.